x-pack/filebeat/input/awss3/sqs_test.go

```diff
@@ -221,6 +223,27 @@ func newSNSSQSMessage() types.Message {
 	}
 }
 
+func newEventBridgeSQSMessage() sqs.Message {
+	body, err := json.Marshal(s3EventsV2{
+		Message: "{\"Messages\":[{ \"version\": \"0\", \"id\": \"f17994c0-7cb9-ee01-79fd-ae46df89e3a4\", \"detail-type\": \"Object Created\", \"source\": \"aws.s3\", \"account\": \"952856826737\", \"time\": \"2024-06-24T08:31:26Z\", \"region\": \"eu-west-1\", \"resources\": [\"arn:aws:s3:::vpc-flow-logs-ks\" ], \"detail\": {\"version\": \"0\",\"bucket\": { \"name\": \"vpc-flow-logs-ks\"},\"object\": { \"key\": \"test-object-key\", \"size\": 744, \"etag\": \"2ba6b152f13c75a9155b87b5b072963c\", \"version-id\": \"uoW5awQhqux4rhi4Nuh6il967FzQlsHJ\", \"sequencer\": \"0066792EC46EC0B037\"},\"request-id\": \"Q93BVWXD5G6FKWC2\",\"requester\": \"516635408544\",\"source-ip-address\": \"10.0.27.95\",\"reason\": \"PutObject\" }}]}",
+	})
+	if err != nil {
+		panic(err)
```

Comment:
Please don't panic here.

Either return an error or accept a `testing.T` so you can cal `Error/Fatal` on it.

Reply:
"[When](https://github.com/elastic/beats/blob/786192b8b599ba4702d7c541aa4f7bde64111d65/x-pack/filebeat/input/awss3/sqs_test.go#L210) in [Rome](https://github.com/elastic/beats/blob/786192b8b599ba4702d7c541aa4f7bde64111d65/x-pack/filebeat/input/awss3/sqs_test.go#L188)"

I'll limit myself to this test class, this patterns seems to have been repeated elsewhere across the codebase

when trying to implement: returning error when the caller can't do anything, other than change the code and recompile, feels a little 'meh'. But since there are other  new...Message functions that take an argument I guess it's ok.

Fixed some warnings while doing this clean up, addressed in https://github.com/elastic/beats/pull/40006/commits/c310ce773a2a1e45cb9f63bcfb465252210d450f